addon/services/file-manager.js

```diff
@@ -136,30 +269,46 @@ export default Ember.Service.extend({
         return new Ember.RSVP.Promise((resolve, reject) => {
             Ember.$.ajax(url, {
                 method,
-                data,
+                data: options.data,
                 processData: false,
                 headers: {
                     Authorization: `Bearer ${accessToken}`
                 },
-                success: (data) => resolve(data),
-                error: (_, __, error) => reject(error)
+                error: (_, __, error) => reject(error),
+                success: (data) => resolve(data)
             });
         });
     },
 
-    _pushToStore(data, id, parentFolder) {
-        // Hack the Waterbutler entity to look like the file Ember expects
-        data.data.id = id;
-        data.data.type = 'file';
-        let attr = data.data.attributes;
-        attr.materializedPath = attr.materialized;
-        attr.dateModified = attr.modified;
-
-        let newFolder = this.get('store').push(data);
+    /**
+     * Force-reload a model from the API.
+     *
+     * @method _reloadModel
+     * @private
+     * @param {Object} model Any Ember model with a `reload` method.
+     * @return {Promise} Promise that resolves to the reloaded model or
+     * rejects with an error message.
+     */
+    _reloadModel(model) {
+        this.set('isReloading', true);
+        return model.reload().then((freshModel) => {
+            this.set('isReloading', false);
```

Comment:
As mentioned above, we'd like to avoid using shared state when possible. One way I can see this being a problem: if `model.reload()` fails, it seems that `then` will never be fired, and `this.isReloading` would never be set back to `false`.

Essentially, all future users of the service might get trapped in the wrong behavior because of a single failed request.

Don't `DS.Model` instances have an `isLoading` attribute on them? Would that do what you need? 
http://emberjs.com/api/data/classes/DS.Model.html#property_isLoading